index.js

```diff
@@ -93,8 +96,22 @@ function walk(node, state) {
   )
 
   if (comments.length) {
-    // @ts-ignore, yes, because they’re nonstandard.
-    node.comments = comments
+    if (state.isEstreeNode) {
+      const leading = []
+      const trailing = []
+      for (const comment of comments) {
+        if (comment.leading) {
+          leading.push(comment)
+        } else {
+          trailing.push(comment)
+        }
+      }
+      node.leadingComments = leading
+      node.trailingComments = trailing
```

Comment:
How about this? Looks more like the rest of the code base?

```suggestion
      node.leadingComments = []
      node.trailingComments = []
      for (const comment of comments) {
        if (comment.leading) {
          node.leadingComments.push(comment)
        } else {
          node.trailingComments.push(comment)
        }
      }
```

---

There are also comments that are neither leading nor trailing, but “inner”. Those are pushed into `trailing` here, is that behavior correct?